Foundation/NSStream.swift

```diff
@@ -202,43 +337,62 @@ public class NSOutputStream : Stream {
     }
     
     public override var streamStatus: Status {
-        return Stream.Status(rawValue: UInt(CFWriteStreamGetStatus(_stream)))!
+        return NSStream.Status(rawValue: UInt(CFWriteStreamGetStatus(_stream)))!
     }
     
     public class func outputStreamToMemory() -> Self {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
+    public func copy(with zone: NSZone?) -> AnyObject{
+        if let url = _url , let shouldAppend = _append {
+            return NSOutputStream(url: url, append: shouldAppend)!
+        } else if let buffer = _buffer, let capacity = _capacity {
+            return NSOutputStream(toBuffer: buffer, capacity: capacity)
+        } else {
+           return  NSOutputStream(toMemory: ())
+        }
+    }
+    
+    public func mutableCopy(with zone: NSZone?) -> AnyObject{
+        return copy(with: nil)
+    }
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
-extension Stream {
+extension NSStream {
     public class func getStreamsToHost(withName hostname: String, port: Int, inputStream: AutoreleasingUnsafeMutablePointer<InputStream?>?, outputStream: AutoreleasingUnsafeMutablePointer<NSOutputStream?>?) {
```

Comment:
Thanks for the feed back i will try to work it in to the PR. How ever i accidentally submitted it un rebased and squashed to apple/master instead of my forks's master. will close and resubmit after